src/Microsoft.Health.Fhir.Core/Features/Definition/SearchParameterDefinitionManager.cs

```diff
@@ -118,19 +119,14 @@ public ValueSets.SearchParamType GetSearchParameterType(SearchParameterInfo sear
 
         public void UpdateSearchParameterHashMap(Dictionary<string, string> updatedSearchParamHashMap)
         {
-            // TODO: Make this thread-safe.
             EnsureArg.IsNotNull(updatedSearchParamHashMap, nameof(updatedSearchParamHashMap));
 
             foreach (KeyValuePair<string, string> kvp in updatedSearchParamHashMap)
             {
-                if (_resourceTypeSearchParameterHashMap.ContainsKey(kvp.Key))
-                {
-                    _resourceTypeSearchParameterHashMap[kvp.Key] = kvp.Value;
-                }
-                else
-                {
-                    _resourceTypeSearchParameterHashMap.Add(kvp.Key, kvp.Value);
-                }
+                _resourceTypeSearchParameterHashMap.AddOrUpdate(
```

Comment:
Just for my understanding, why/where is this being called concurrently from?

Reply:
Once we implement Custom Search, it would be possible for multiple requests to update/add search parameters to come in at the same time.  Right now, this is only called during start up when we build the search parameter definitions.  @namadabu please correct me if I am wrong.

Reply:
@brendankowitz +1 to what @feordin mentioned,

Reply:
Agree this change is needed. My question was just why its potentially being called multiple times during startup